File: scripts/generate_galleries.py
#!/usr/bin/env python3
"""
Small generator that injects <img> tags into galeria.html and index.html
between markers. Use if Node isn't available.
Run: python3 scripts/generate_galleries.py
"""
import os
from pathlib import Path

root = Path(__file__).resolve().parent.parent
gallery_dir = root / 'assets' / 'images' / 'gallery'
home_dir = root / 'assets' / 'images' / 'home'

gal_html = root / 'galeria.html'
index_html = root / 'index.html'

def make_imgs(dirpath, webprefix):
    if not dirpath.exists():
        return []
    files = [f for f in sorted(dirpath.iterdir()) if f.suffix.lower() in ['.jpg','.jpeg','.png','.gif','.webp']]
    return [f"{webprefix}/{f.name}" for f in files]

def inject(file_path, start_marker, end_marker, items_html):
    s = file_path.read_text(encoding='utf8')
    si = s.find(start_marker)
    ei = s.find(end_marker, si)
    if si == -1 or ei == -1:
        print(f"Markers not found in {file_path}")
        return
    before = s[:si+len(start_marker)]
    after = s[ei:]
    new = before + '\n' + '\n'.join(items_html) + '\n' + after
    file_path.write_text(new, encoding='utf8')
    print(f"Updated {file_path}")


gallery_imgs = make_imgs(gallery_dir, '/assets/images/gallery')
home_imgs = make_imgs(home_dir, '/assets/images/home')

gallery_html = [f'          <img src="{src}" alt="Galerie {i+1}">' for i,src in enumerate(gallery_imgs)]
home_html = [f'            <img src="{src}" alt="Momentka {i+1}">' for i,src in enumerate(home_imgs)]

inject(gal_html, '<!-- GALLERY:START -->', '<!-- GALLERY:END -->', gallery_html if gallery_html else ['          <!-- no images found -->'])
inject(index_html, '<!-- HOMEGALLERY:START -->', '<!-- HOMEGALLERY:END -->', home_html if home_html else ['            <!-- no images found -->'])

print('Done')

File: scripts/generate_galleries.js
/*
  Simple generator: scans assets/images/gallery and assets/images/home
  and injects <img> tags into galeria.html (between <!-- GALLERY:START --> markers)
  and index.html (between <!-- HOMEGALLERY:START --> markers).

  Usage: node scripts/generate_galleries.js
*/
const fs = require('fs');
const path = require('path');

const root = path.resolve(__dirname, '..');
const galleryDir = path.join(root, 'assets', 'images', 'gallery');
const homeDir = path.join(root, 'assets', 'images', 'home');

const galHtml = path.join(root, 'galeria.html');
const indexHtml = path.join(root, 'index.html');

function makeImgs(dir, webPrefix){
  if(!fs.existsSync(dir)) return [];
  return fs.readdirSync(dir)
    .filter(f=>/[.](jpe?g|png|gif|webp)$/i.test(f))
    .map(f=>`${webPrefix}/${f}`);
}

function inject(filePath, startMarker, endMarker, itemsHtml){
  let content = fs.readFileSync(filePath,'utf8');
  const start = content.indexOf(startMarker);
  const end = content.indexOf(endMarker, start);
  if(start===-1 || end===-1){
    console.error('Markers not found in', filePath);
    return;
  }
  const before = content.slice(0, start+startMarker.length);
  const after = content.slice(end);
  const newContent = before + '\n' + itemsHtml.join('\n') + '\n' + after;
  fs.writeFileSync(filePath, newContent, 'utf8');
  console.log('Updated', filePath);
}

// Build html arrays
const galleryImgs = makeImgs(galleryDir, '/assets/images/gallery');
const homeImgs = makeImgs(homeDir, '/assets/images/home');

const galleryHtml = galleryImgs.map((src,i)=>`          <img src="${src}" alt="Galerie ${i+1}">`);
const homeHtml = homeImgs.map((src,i)=>`            <img src="${src}" alt="Momentka ${i+1}">`);

// Inject into files
inject(galHtml, '<!-- GALLERY:START -->', '<!-- GALLERY:END -->', galleryHtml.length?galleryHtml:['          <!-- no images found -->']);
inject(indexHtml, '<!-- HOMEGALLERY:START -->', '<!-- HOMEGALLERY:END -->', homeHtml.length?homeHtml:['            <!-- no images found -->']);

console.log('Done');
